### PR TITLE
Add relevant parameters for antismash

### PR DESCRIPTION
**Pull request by Darcy220606**
## PR checklist : This is to address issue #80 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/funcscan/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/funcscan _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -431,6 +431,52 @@
...
+                    "type": "integer",
+                    "default": 1000,
+                    "description": "Only process sequences larger than <minlength> (default: 1000).",
+                    "fa_icon": "fas fa-align-center"
+                },
+                "bgc_antismash_hmmdetectionstrictness": {
+                    "type": "string",
+                    "default": "relaxed",
+                    "description": "Defines which level of strictness to use for HMM-based cluster detection",
+                    "help_text": "Defines which level of strictness to use for HMM-based cluster detection. Default is set at 'relaxed'. Other valid options can be 'strict' or 'loose'.",
```
Comment (jasmezz) [inline nextflow_schema.json:469]:
```suggestion
                    "help_text": "Defines which level of strictness to use for HMM-based cluster detection. Default is set at 'relaxed'. Other valid options are 'strict' or 'loose'.",
```

File: nextflow_schema.json [hunk 2]
```diff
@@ -431,6 +431,52 @@
...
+                    "fa_icon": "fas fa-search",
+                    "enum": ["relaxed", "strict", "loose"]
+                },
+                "bgc_antismash_taxon": {
+                    "type": "string",
+                    "default": "bacteria",
+                    "description": "Taxonomic classification of input sequence.",
+                    "help_text": "Taxonomic classification of input sequence. The default is set at 'bacteria'. Another valid option is 'fungi'.",
```
Comment (jasmezz) [inline nextflow_schema.json:477]:
```suggestion
                    "help_text": "Taxonomic classification of input sequence. Valid options are 'bacteria' (default) and 'fungi'.",
```

File: conf/modules.config [hunk 3]
```diff
@@ -251,7 +251,17 @@ process {
...
             saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
         ]
-        ext.args   = { "--allow-long-headers" }
+        ext.args   = [
+            params.bgc_antismash_cbgeneral ? '--cb-general' : '',
+            params.bgc_antismash_cbknownclusters ? '--cb-knownclusters' : '',
+            params.bgc_antismash_smcogtrees ? '--smcog-trees' : '',
+            params.bgc_antismash_ccmibig ? '--cc-mibig' : '',
+            params.bgc_antismash_cbsubclusters ? '--cb-subclusters' : '',
+            "--allow-long-headers",
```
Comment (jasmezz) [inline conf/modules.config:260]: Yes!! Very important 🥴

File: nextflow.config [hunk 4]
```diff
@@ -76,9 +76,17 @@ params {
     // BGC options
     run_bgc_screening                   = false
```
Comment (jasmezz) [inline nextflow.config:77]:
```suggestion
    run_bgc_screening                      = false
```

**Review by jasmezz — CHANGES_REQUESTED** [review]
Very nice! So there is no option to let the user add more arguments than we offer in the paramters (since your PR doesn't list everything possible). Or is it too risky? If that's the case, let's just add every tiny argument listed in the manual (`antismash --help-showall`), like `--cb-min-homology-scale`?

File: nextflow_schema.json [hunk 5]
```diff
@@ -431,6 +431,52 @@
                     "description": "Path to user-defined local antiSMASH directory",
                     "fa_icon": "far fa-folder-open",
                     "default": "None"
+                },
+                "bgc_antismash_cbgeneral": {
+                    "type": "boolean",
+                    "description": "Compare identified clusters against a database of antiSMASH-predicted clusters."
```
Comment (louperelo) [inline nextflow_schema.json:437]: Add a fa_icon, i.e:
```suggestion
                    "description": "Compare identified clusters against a database of antiSMASH-predicted clusters.",
                    "fa_icon": "fab fa-affiliatetheme"
```

**Review by louperelo — APPROVED** [review]
Looks good to go, if the tests passed locally. BGC tests continue with the db-download problems in the BGC workflow.

**Comment by Darcy220606** [comment]
> Very nice! So there is no option to let the user add more arguments than we offer in the paramters (since your PR doesn't list everything possible). Or is it too risky? If that's the case, let's just add every tiny argument listed in the manual (`antismash --help-showall`), like `--cb-min-homology-scale` etc.?

Indeed @jasmezz - I tried to pick only the paparametrs that would not affect the output. Particularly that we dont want it to interfere with the comBGC file format requirements. There are also some parametrs that are only for the `cluster` option when invoked, so i think we should leave them to the user if they want to add them then in the config file.

File: nextflow.config [hunk 6]
```diff
@@ -74,11 +74,19 @@ params {
     arg_abricate_db                     = 'card'
 
     // BGC options
-    run_bgc_screening                   = false
-
-    bgc_skip_antismash                  = false
-    bgc_antismash_databases             = null
-    bgc_antismash_installationdirectory = null
+    run_bgc_screening                      = false
```
Comment (jasmezz) [inline nextflow.config:77]:
```suggestion
    run_bgc_screening                       = false
```

**Review by jasmezz — APPROVED** [review]
:+1:

File: nextflow_schema.json [hunk 7]
```diff
@@ -437,6 +437,53 @@
                     "default": 1000,
                     "description": "Minimum contig length a sample needs to have to be run",
                     "fa_icon": "fas fa-ruler-horizontal"
+                },
+                "bgc_antismash_cbgeneral": {
+                    "type": "boolean",
+                    "description": "Compare identified clusters against a database of antiSMASH-predicted clusters.",
```
Comment (jfy133) [inline nextflow_schema.json:443]:
```suggestion
                    "description": "Compare identified clusters against a database of antiSMASH-predicted clusters.",
                    "help": "Compare identified clusters against a database of antiSMASH-predicted clusters.\n\n> Modifies tool parameter(s)\n> antiSMASH: `--cb-general`\n"
```

**Review by jfy133 — APPROVED** [review]
You have my approval (as I just updated most of them ; ) ), but please review my changes @Darcy220606 ! Merge in if you're happy otherwise make your changes and re-request review from me .

File: nextflow.config [hunk 8]
```diff
@@ -74,12 +74,20 @@ params {
...
+    bgc_skip_antismash                      = false
+    bgc_antismash_databases                 = null
+    bgc_antismash_installationdirectory     = null
+    bgc_antismash_cbgeneral                 = false
+    bgc_antismash_cbknownclusters           = false
+    bgc_antismash_cbsubclusters             = false
+    bgc_antismash_smcogtrees                = false
+    bgc_antismash_ccmibig                   = false
+    bgc_antismash_contigminlength                 = 1000
+    bgc_antismash_hmmdetectionstrictness    = 'relaxed'
```
Comment (Darcy220606) [inline nextflow.config:88]:
```suggestion
    bgc_antismash_ccmibig                   = false
    bgc_antismash_contigminlength           = 1000
    bgc_antismash_hmmdetectionstrictness    = 'relaxed'
```

File: nextflow.config [hunk 9]
```diff
@@ -74,12 +74,20 @@ params {
...
+
+    bgc_skip_antismash                      = false
+    bgc_antismash_databases                 = null
+    bgc_antismash_installationdirectory     = null
+    bgc_antismash_cbgeneral                 = false
+    bgc_antismash_cbknownclusters           = false
+    bgc_antismash_cbsubclusters             = false
+    bgc_antismash_smcogtrees                = false
+    bgc_antismash_ccmibig                   = false
+    bgc_antismash_contigminlength                 = 1000
```
Comment (jasmezz) [inline nextflow.config:87]:
```suggestion
    bgc_antismash_contigminlength           = 1000
```

File: nextflow_schema.json [hunk 10]
```diff
@@ -10,7 +10,10 @@
             "type": "object",
             "fa_icon": "fas fa-terminal",
             "description": "Define where the pipeline should find input data and save output data.",
-            "required": ["input", "outdir"],
+            "required": [
+                "input",
+                "outdir"
+            ],
```
Comment (Darcy220606) [inline nextflow_schema.json:16]: Im not sure, but i think in this format, `prettier` still needs to be run on the schema.

File: nextflow_schema.json [hunk 11]
```diff
@@ -422,21 +435,84 @@
...
+                        "relaxed",
+                        "strict",
+                        "loose"
+                    ]
+                },
+                "bgc_antismash_taxon": {
+                    "type": "string",
+                    "default": "bacteria",
+                    "description": "Specify which taxonomic classification of input sequence to use",
+                    "help_text": "This species which set of secondary metabolites to screen for, based on the taxon type the secondary metabolites are from.\n\nThis will run different pipelines depending on whether the input sequences are from bacteria or fungi.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--taxon`",
```
Comment (Darcy220606) [inline nextflow_schema.json:510]:
```suggestion
                    "help_text": "This specifes which set of secondary metabolites to screen for, based on the taxon type the secondary metabolites are from.\n\nThis will run different pipelines depending on whether the input sequences are from bacteria or fungi.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--taxon`",
```

**Review by Darcy220606 — COMMENTED** [review]
Nice! Very mush elaborated help-text :)

File: nextflow_schema.json (hunk 10 above)
Comment (jfy133) [reply nextflow_schema.json:16]: Will do!

File: nextflow_schema.json [hunk 12]
```diff
@@ -422,21 +435,84 @@
...
+                },
+                "bgc_antismash_cbsubclusters": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against known subclusters responsible for synthesising precursors.",
+                    "fa_icon": "fas fa-adjust",
+                    "help_text": "Turns on additional screening for operons involved in the biosynthesis of early secondary metabolites components using the cluster blast algorithm.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-subclusters`"
+                },
+                "bgc_antismash_ccmibig": {
+                    "type": "boolean",
+                    "description": "Turns on ClusterCompare comparison against known gene clusters from the MIBiG database.",
```
Comment (jfy133) [inline nextflow_schema.json:484]:
```suggestion
                    "description": "Turn on ClusterCompare comparison against known gene clusters from the MIBiG database.",
```

File: nextflow_schema.json [hunk 13]
```diff
@@ -422,21 +435,84 @@
...
+                },
+                "bgc_antismash_cbknownclusters": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against known gene clusters from the MIBiG database.",
+                    "fa_icon": "fas fa-puzzle-piece",
+                    "help_text": "This will turn on comparing identified clusters against known gene clusters from the MIBiG database using the cluster blast algorithm.\n\n[MIBiG](https://mibig.secondarymetabolites.org/) is a curated datbase of experimentally characterised gene clusters and with rich associated metadata.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-knownclusters`\n"
+                },
+                "bgc_antismash_cbsubclusters": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against known subclusters responsible for synthesising precursors.",
```
Comment (jfy133) [inline nextflow_schema.json:478]:
```suggestion
                    "description": "Turn on clusterblast comparison against known subclusters responsible for synthesising precursors.",
```

File: nextflow_schema.json [hunk 14]
```diff
@@ -422,21 +435,84 @@
...
+                },
+                "bgc_antismash_cbgeneral": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against database of antiSMASH-predicted clusters.",
+                    "help_text": "Compare identified clusters against a database of antiSMASH-predicted clusters using the cluster blast algorithm.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-general`\n",
+                    "fa_icon": "fab fa-affiliatetheme"
+                },
+                "bgc_antismash_cbknownclusters": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against known gene clusters from the MIBiG database.",
```
Comment (jfy133) [inline nextflow_schema.json:472]:
```suggestion
                    "description": "Turn on clusterblast comparison against known gene clusters from the MIBiG database.",
```

File: nextflow_schema.json [hunk 15]
```diff
@@ -422,21 +435,84 @@
...
+                "bgc_antismash_contigminlength": {
+                    "type": "integer",
+                    "default": 1000,
+                    "description": "Minimum length a contig must have to be screened with antiSMASH.",
+                    "fa_icon": "fas fa-align-center",
+                    "help_text": "This specifies the minimum length that a contig must have for the contig to be screened by antiSMASH.\n\nThis will only apply to samples that are screened with antiSMASH (i.e., those samples that have not been removed by `--bgc_antismash_sampleminlength`).\n\nYou may wish to increase this value compared to that of `--bgc_antismash_sampleminlength`, in cases where you wish to screen higher-quality (i.e., longer) contigs, or speed up runs by not screening lower quality/less informative contigs.\n\n> Modifies tool parameter(s):\n> - --minlength"
+                },
+                "bgc_antismash_cbgeneral": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against database of antiSMASH-predicted clusters.",
```
Comment (jfy133) [inline nextflow_schema.json:466]:
```suggestion
                    "description": "Turn on clusterblast comparison against database of antiSMASH-predicted clusters.",
```

File: nextflow_schema.json [hunk 16]
```diff
@@ -422,21 +435,84 @@
...
+                    "help_text": "This will turn on comparing identified clusters against known gene clusters from the MIBiG database using the cluster blast algorithm.\n\n[MIBiG](https://mibig.secondarymetabolites.org/) is a curated datbase of experimentally characterised gene clusters and with rich associated metadata.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-knownclusters`\n"
+                },
+                "bgc_antismash_cbsubclusters": {
+                    "type": "boolean",
+                    "description": "Turns on clusterblast comparison against known subclusters responsible for synthesising precursors.",
+                    "fa_icon": "fas fa-adjust",
+                    "help_text": "Turns on additional screening for operons involved in the biosynthesis of early secondary metabolites components using the cluster blast algorithm.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-subclusters`"
```
Comment (jfy133) [inline nextflow_schema.json:480]:
```suggestion
                    "help_text": "Turn on additional screening for operons involved in the biosynthesis of early secondary metabolites components using the cluster blast algorithm.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cb-subclusters`"
```

File: nextflow_schema.json [hunk 17]
```diff
@@ -422,21 +435,84 @@
...
+                },
+                "bgc_antismash_ccmibig": {
+                    "type": "boolean",
+                    "description": "Turns on ClusterCompare comparison against known gene clusters from the MIBiG database.",
+                    "fa_icon": "fab fa-affiliatetheme",
+                    "help_text": "Turns on comparison of detected genes against the MIBiG database using the ClusterCompare algorithm - an alternative to clusterblast.\n\nNote there will not be a dedicated ClusterCompare output in the antiSMASH results directory, but is present in the HTML.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cc-mibig`"
```
Comment (jfy133) [inline nextflow_schema.json:486]:
```suggestion
                    "help_text": "Turn on comparison of detected genes against the MIBiG database using the ClusterCompare algorithm - an alternative to clusterblast.\n\nNote there will not be a dedicated ClusterCompare output in the antiSMASH results directory, but is present in the HTML.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\n> Modifies tool parameter(s):\n> - antiSMASH: `--cc-mibig`"
```

File: nextflow_schema.json [hunk 18]
```diff
@@ -486,10 +562,10 @@
                 },
                 "bgc_hmmsearch_models": {
                     "type": "string",
-                    "default": "None",
                     "help_text": "HMMSearch performs biosequence analysis using profile hidden Markov Models.\nThe models are specified in`.hmm` files that are specified with this parameter\n\ne.g. \n\n```\n--bgc_hmmsearch_models '/<path>/<to>/<models>/*.hmm'\n```\n\nYou must wrap the path in quotes if you use a wilcard!\n",
                     "fa_icon": "fas fa-layer-group",
-                    "description": "Specify path (wrapped in quotes if using wildcard!) to the BGC hmm model files to search against."
+                    "description": "Specify path (wrapped in quotes if using wildcard!) to the BGC hmm model files to search against.",
```
Comment (jfy133) [inline nextflow_schema.json:567]: See how I've phrased this in #182 for AMPs and use this here too

**Review by jfy133 — CHANGES_REQUESTED** [review]

File: nextflow_schema.json [hunk 19]
```diff
@@ -642,21 +652,84 @@
...
+                },
+                "bgc_antismash_contigminlength": {
+                    "type": "integer",
+                    "default": 1000,
+                    "description": "Minimum length a contig must have to be screened with antiSMASH.",
+                    "fa_icon": "fas fa-align-center",
+                    "help_text": "This specifies the minimum length that a contig must have for the contig to be screened by antiSMASH.\n\nThis will only apply to samples that are screened with antiSMASH (i.e., those samples that have not been removed by `--bgc_antismash_sampleminlength`).\n\nYou may wish to increase this value compared to that of `--bgc_antismash_sampleminlength`, in cases where you wish to screen higher-quality (i.e., longer) contigs, or speed up runs by not screening lower quality/less informative contigs.\n\n> Modifies tool parameter(s):\n> - --minlength"
```
Comment (jfy133) [inline nextflow_schema.json:679]:
```suggestion
                    "help_text": "This specifies the minimum length that a contig must have for the contig to be screened by antiSMASH.\n\nFor more information see the [antiSMASH documentation](https://docs.antismash.secondarymetabolites.org/).\n\nThis will only apply to samples that are screened with antiSMASH (i.e., those samples that have not been removed by `--bgc_antismash_sampleminlength`).\n\nYou may wish to increase this value compared to that of `--bgc_antismash_sampleminlength`, in cases where you wish to screen higher-quality (i.e., longer) contigs, or speed up runs by not screening lower quality/less informative contigs.\n\n> Modifies tool parameter(s):\n> - --minlength"
```

File: subworkflows/local/bgc.nf [hunk 20]
```diff
@@ -64,8 +64,8 @@ workflow BGC {
                                 .groupTuple()
                                 .filter {
                                     meta, files ->
-                                        if ( meta.longest_contig < params.bgc_antismash_mincontiglength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of --bgc_antismash_mincontiglength ${params.bgc_antismash_mincontiglength}. Antismash will not be run for: ${meta.id}."
-                                        meta.longest_contig >= params.bgc_antismash_mincontiglength
+                                        if ( meta.longest_contig < params.bgc_antismash_sampleminlength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of --bgc_antismash_sampleminlength ${params.bgc_antismash_sampleminlength}. Antismash will not be run for: ${meta.id}."
```
Comment (jfy133) [inline subworkflows/local/bgc.nf:67]:
```suggestion
                                        if ( meta.longest_contig < params.bgc_antismash_sampleminlength ) log.warn "[nf-core/funcscan] Sample does not have any contig reaching min. length threshold of --bgc_antismash_sampleminlength ${params.bgc_antismash_sampleminlength}. Antismash will not be run for sample: ${meta.id}."
```

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Pull request merged**